src/js/actions/export.js

```diff
@@ -119,6 +120,40 @@ define(function (require, exports) {
     };
 
     /**
+     * Get current status of generator
+     * @private
+     * @return {Promise.<boolean>}
+     */
+    var _getGeneratorStatus = function () {
+        return descriptor.playObject(generatorLib.getGeneratorStatus())
+            .then(function (status) {
+                return objUtil.getPath(status, "generatorStatus.generatorStatus") === 1;
+            });
```

Comment:
As written, if _getGeneratorStatus() rejects, it will result in giving up on connecting to the service.  I guess your point is that we should swallow these, and try to enable generator anyway.  I can go with that.